libraries/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import {
+  FileSystem,
+  AlreadyReportedError,
+  Executable,
+  EnvironmentMap,
+  ITerminal,
+  Colors,
+  Terminal,
+  ConsoleTerminalProvider
+} from '@rushstack/node-core-library';
+import { PrintUtilities } from '@rushstack/terminal';
+
+import { RushConfiguration } from '../api/RushConfiguration';
+import { NodeJsCompatibility } from '../logic/NodeJsCompatibility';
+import { SpawnSyncReturns } from 'child_process';
+import { ILaunchOptions } from '../api/Rush';
+
+export interface ILaunchRushPnpmInternalOptions extends ILaunchOptions {}
+
+const RUSH_SKIP_CHECKS_PARAMETER: string = '--rush-skip-checks';
+
+enum CommandKind {
+  KnownSafe,
+  ShowWarning,
+  Blocked
+}
```

Comment:
(@dmichon-msft but isn't a string comparison generally more expensive than an integer comparison? In general -- performance doesn't matter at all in this case)

Reply:
If we were concerned about performance, the more logical thing to do would be to have the value in the map directly be the relevant handler, instead of map -> switch/case

Reply:
Actually, @chengcyber , could you just inline the map directly into the switch/case? Rather than doing two lookups just switch on the command name and have several cases map to each handler.